test/functional/wallet_simulaterawtx.py

```diff
@@ -0,0 +1,99 @@
+        node.createwallet(wallet_name='w1')
+        w0 = node.get_wallet_rpc('w0')
+        w1 = node.get_wallet_rpc('w1')
+
+        self.generatetoaddress(node, COINBASE_MATURITY + 1, w0.getnewaddress())
+        assert_equal(w0.getbalance(), 50.0)
+        assert_equal(w1.getbalance(), 0.0)
+
+        address1 = w1.getnewaddress()
+        address2 = w1.getnewaddress()
+
+        # Make address1 watch-only in w0
+        w0.importpubkey(pubkey=w1.getaddressinfo(address1)["pubkey"])
+        all = {"include_watchonly": True}
+
+        tx1 = node.createrawtransaction([], [{address1: 5.0}])
+        tx2 = node.createrawtransaction([], [{address2: 10.0}])
+
+        # w0 should be unaffected (include_watchonly=False) or see +5 (include_watchonly=True) for tx1
+        assert_equal(w0.simulaterawtransaction([tx1])["balance_change"], 0.0)
+        assert_equal(w0.simulaterawtransaction([tx1], all)["balance_change"], 5.0)
+
+        # w1 should see +5 balance for tx1
+        assert_equal(w1.simulaterawtransaction([tx1])["balance_change"], 5.0)
+
+        # w0 should be unaffected (include_watchonly=False) or see +5 (include_watchonly=True) for both transactions
+        assert_equal(w0.simulaterawtransaction([tx1, tx2])["balance_change"], 0.0)
+        assert_equal(w0.simulaterawtransaction([tx1, tx2], all)["balance_change"], 5.0)
+
+        # w1 should see +15 balance for both transactions
+        assert_equal(w1.simulaterawtransaction([tx1, tx2])["balance_change"], 15.0)
+
+        # w0 funds transaction; it should now see a decrease in (tx fee and payment), and w1 should see the same as above
+        funding = w0.fundrawtransaction(tx1)
+        tx1 = funding["hex"]
+        bitcoin_fee = Decimal(funding["fee"])
+
+        # w0 sees fee + 5 btc decrease, or fee only for watchonly case
+        assert_approx(w0.simulaterawtransaction([tx1])["balance_change"], -(Decimal("5") + bitcoin_fee))
+        assert_approx(w0.simulaterawtransaction([tx1], all)["balance_change"], -(bitcoin_fee))
+
+        # w1 sees same as before
+        assert_equal(w1.simulaterawtransaction([tx1])["balance_change"], 5.0)
+
+        # same inputs (tx) more than once should error
+        assert_raises_rpc_error(-8, "Transaction(s) are spending the same output more than once", w0.simulaterawtransaction, [tx1,tx1])
+
+        # send tx1 to avoid reusing same UTXO below
+        node.sendrawtransaction(w0.signrawtransactionwithwallet(tx1)["hex"])
+        self.sync_all()
+
+        # w0 funds transaction 2; it should now see a decrease in (tx fee and payment), and w1 should see the same as above
+        funding = w0.fundrawtransaction(tx2)
+        tx2 = funding["hex"]
+        bitcoin_fee2 = Decimal(funding["fee"])
+
+        # w0 sees fees + 15 btc decrease, or fees + 10 btc decrease for watch only case
+        assert_approx(w0.simulaterawtransaction([tx1, tx2])["balance_change"], -(Decimal("5") + bitcoin_fee + Decimal("10") + bitcoin_fee2))
```

Comment:
In ae8564c198a3f9c57cc896489afcb78d6b947b45 "rpc/wallet: add simulaterawtransaction RPC"

Since `tx1` is already sent, it will include inputs that are already spent. It seems like we should disallow transactions that spend already spent inputs, at least until we implement conflict checking in this RPC.

Reply:
Added fetch/check of coins for transactions and RPC asserts when coins are missing or spent.